### ISSUE TITLE
Usage with handlebars?

### ISSUE THREAD

**Issue by louis030195** [uiwjs/react-codemirror#333]
Hello!

It's not so clear how to use this lib with handlebars, if it is even possible, I see codemirror support it so why not?

I couldn't find it in `import { handlebars } from '@codemirror/legacy-modes/mode/handlebars';`?

Thanks a lot

**Comment by jaywcjlove** [comment]
@louis030195 

```jsx
import CodeMirror from '@uiw/react-codemirror';
import { StreamLanguage } from '@codemirror/language';
import { handlebars } from '@codemirror/legacy-modes/mode/handlebars';

const codeStr = `code string....`;

export default function App() {
  return (
    <CodeMirror
      value={codeStr}
      height="200px"
      extensions={[StreamLanguage.define(handlebars)]}
      onChange={(value, viewUpdate) => {
        console.log('value:', value);
      }}
    />
  );
}
```

**Comment by louis030195** [comment]
> ```js
> extensions={[StreamLanguage.define(handlebars)]}
> ```

Module not found: Error: Can't resolve '@codemirror/legacy-modes/mode/handlebars' ?

`package.json`
```json
...
  "dependencies": {
    ...
    "@codemirror/language": "^6.0.0",
    "@codemirror/legacy-modes": "^6.0.0",
...
```

**Comment by Miltonbhowmick** [comment]
> @louis030195
> 
> ```js
> import CodeMirror from '@uiw/react-codemirror';
> import { StreamLanguage } from '@codemirror/language';
> import { handlebars } from '@codemirror/legacy-modes/mode/handlebars';
> 
> const codeStr = `code string....`;
> 
> export default function App() {
>   return (
>     <CodeMirror
>       value={codeStr}
>       height="200px"
>       extensions={[StreamLanguage.define(handlebars)]}
>       onChange={(value, viewUpdate) => {
>         console.log('value:', value);
>       }}
>     />
>   );
> }
> ```

This solves my problem!

**Comment by tslater** [comment]
@louis030195 I'm having the same issue. Did you ever figure out how to import?

**Comment by lenqwang** [comment]
Is seems v6 does not support handlebars, so how can v6 support handlebars?

**Comment by jaywcjlove** [comment]
@tslater @lenqwang https://github.com/uiwjs/react-codemirror/issues/333#issuecomment-1238887212

**Comment by lenqwang** [comment]
> @tslater @lenqwang https://github.com/uiwjs/react-codemirror/issues/333#issuecomment-1238887212

I don't find the handlebars module file

**Comment by tslater** [comment]
I don't think there is any good way to use it on v6 yet. You'd have to implement it yourself.

**Comment by jaywcjlove** [comment]
@lenqwang  @tslater https://codemirror.net/5/mode/handlebars/index.html Is it this one? I think the definition is also very simple.

**Comment by lenqwang** [comment]
Yes, I had to implement it myself. Although I searched the entire internet.
However, I am very curious why the official has not provided an implementation of handlebars.

**Comment by jaywcjlove** [comment]
- https://discuss.codemirror.net/t/handlebars-in-v6/5941
- https://stackblitz.com/edit/node-rwuxbu?file=src%2Fsyntax.grammar,src%2Fhighlight.ts
- https://codemirror.net/examples/mixed-language/

**Comment by lenqwang** [comment]
Thanks a lot, These links are very useful to me
